src/databricks/sql/parameters/choose.py

```diff
@@ -0,0 +1,80 @@
+from databricks.sql.parameters.native import ParameterApproach
+from typing import Optional, List, Tuple, TYPE_CHECKING
+
+
+if TYPE_CHECKING:
+    from databricks.sql.client import Connection
+    from databricks.sql.parameters.native import TParameterCollection, TSparkParameter
+
+
+from databricks.sql.parameters.native import (
+    _normalize_tparametercollection,
+    _determine_parameter_structure,
+    prepare_native_parameters,
+)
+from databricks.sql.parameters.inline import (
+    transform_paramstyle,
+    prepare_inline_parameters,
+)
+
+
+NO_NATIVE_PARAMS: List = []
+
+
+def determine_parameter_approach(
+    connection: "Connection", params: Optional["TParameterCollection"]
+) -> ParameterApproach:
+    """Encapsulates the logic for choosing whether to send parameters in native vs inline mode
+
+    If params is None then ParameterApproach.NONE is returned.
+    If self.use_inline_params is True then inline mode is used.
+    If self.use_inline_params is False, then check if the server supports them and proceed.
+        Else raise an exception.
```

Comment:
Although I believe this is just a refactor and not actually new code, I noticed something:

We claim that if self.use_inline_params is False, we check if the server supports them and proceed, else raise an exception.  None of that is in the current function.  This is precisely why I recommend not detailing how a function works in the doc string, as it is a maintenance burden.  Recommend one sentence on what a function does, and no discussion of how it does.  We don't want other code to depend on the how, just the what.

Reply:
Fixed!